### PR TITLE
Smuggler 5.6

### PR DESCRIPTION
**Pull request by jburel**
This brings smuggler to 5.6 gateway

To build and deploy http://c0c0n3.github.io/ome-smuggler/docs/content/examples/whirlwind-tour.html

cc @c0c0n3 

To test:

* open a terminal and checkout this PR
*  docker build -t smuggler .
* docker run -it  -p 8000:8000 smuggler 
* Make sure the user has an email address set up or use the daily build of OMERO.insight
* in ``insight/config/container.xml`` Edit ``<entry name="omero.client.import.offline.enabled" type="boolean">true</entry>``
* Import an  image using insight. Check the output in smuggler. Ignore failure when sending email. This is not set up.

### CODE REVIEW

**Comment by c0c0n3** [comment]
Looks good to me. Just a couple of questions tough.

1. Should we do some kind of versioning? For example, [this](https://github.com/c0c0n3/ome-smuggler/pull/15/commits/779307651e4e8dbab44ff39a51f3f53247953f59) is what I had in Smuggler. BTW files & version numbers are still there, but I don't know how meaningful they will be now in the context of OMERA?
2. How should we release it? [This](https://github.com/c0c0n3/ome-odd-n-ends/tree/master/smugs-at-igh) was our release & deployment procedure at IGH back in the day. Not sure what you guys would like to do going forward...

This PR resolves #12, so I'm going to ditch my [OMERO upgrade branch](https://github.com/c0c0n3/omero-ms-queue/commit/df29c21f2c3d4a2cf41c3a2b72253390460e6fe0).

cc @jburel @juliomateoslangerak

**Comment by jburel** [comment]
@c0c0n3 I have modified the gradle file in order to work with 5.5
I have so far not been able to import
``"sessionKey:•9-30d6-44a4-90b2-8b8c083631de"`` always seems to be truncated when they are used in smuggler.

**Comment by jburel** [comment]
error so it seems to come from
```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: java.lang.NoClassDefFoundError: gnu/getopt/LongOpt
	at ome.formats.importer.cli.CommandLineImporter.main(CommandLineImporter.java:556)
	at ome.cli.omero.imports.ImportAdapter.main(ImportAdapter.java:14)
	at ome.cli.Main.main(Main.java:36)
	... 8 more
Caused by: java.lang.ClassNotFoundException: gnu.getopt.LongOpt
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:94)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

**Comment by jburel** [comment]
Now I can import from script but not insight yet!!

**Comment by jburel** [comment]
I can now import from insight

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ARG COMPONENT=server
+ARG RUN_IMAGE=openmicroscopy/omero-${COMPONENT}:latest
+
+
+FROM ${BUILD_IMAGE} as build
+USER root
+RUN apt-get update \
+ && apt-get install -y ant gradle maven vim
+
+# openjdk:8 is "stretch" or Debian 9
```
Comment (mtbc) [inline Dockerfile]: I don't see why this comment is here.

Comment (jburel) [reply Dockerfile]: copy/paste from another file

**Comment by jburel** [comment]
I have tested against a 5.6.dev server and outreach (5.5)

File: components/cli/cli.gradle [hunk 2]
```diff
@@ -14,18 +14,12 @@ repositories {
...
 dependencies {
-    compile(group: 'ome', name: 'formats-gpl', version: '5.5.0'){
-    }
-    compile(group: 'ome', name: 'formats-bsd', version: '5.5.0'){
-    }
-    compile(group: 'org.springframework.ldap', name: 'spring-ldap', version: '1.3.0.RELEASE', classifier: 'all'){
-        exclude group: 'org.testng', module: 'testng'
-    }
-    compile(group: 'omero', name: 'blitz', version: '5.3.2-ice36-b62') {
+    compile(group: 'org.openmicroscopy', name: 'omero-gateway', version: '5.5.4') {
```
Comment (joshmoore) [inline components/cli/cli.gradle]:
```suggestion
    compile(group: 'org.openmicroscopy', name: 'omero-gateway', version: '5.6.2') {
```

Comment (jburel) [reply components/cli/cli.gradle]: I made the change locally and run with 5.6.2 but I forgot to push the commit :-)

**Comment by joshmoore** [comment]
LGTM. (I assume along with ome/openmicroscopy's `Dockerfile` we could simplify this one when the time comes).

**Pull request merged**